Microsoft.NET.Build.Containers/Image.cs

```diff
@@ -84,11 +110,35 @@ private static HashSet<Label> ReadLabelsFromConfig(JsonNode inputConfig)
         }
         else
         {
-            // initialize and empty labels map
+            // initialize an empty labels map
             return new HashSet<Label>();
         }
     }
 
+    private static HashSet<Port> ReadPortsFromConfig(JsonNode inputConfig)
+    {
+        if (inputConfig is JsonObject config && config["ExposedPorts"] is JsonObject portsJson)
+        {
+            // read label mappings from object
+            var ports = new HashSet<Port>();
```

Comment:
Think it'll be a problem that HashSet isn't order-preserving? It doesn't seem like it would be to me . . .

Reply:
Agree - the underlying collection (a JS object) isn't ordered either